File: problems/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
from typing import Dict, List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        has: Dict[int, int] = {}
        for i in range(len(nums)):
            num = nums[i]
            if num in has:
                return [has[num], i]
            need = target - num
            has[need] = i
        raise ValueError("No two sum solution")


# @lc code=end

if __name__ == "__main__":
    print(Solution().twoSum([2, 7, 11, 15], 9))

File: problems/1.two-sum.kt
package p1

class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = mutableMapOf<Int, Int>()
        nums.forEachIndexed { index, i ->
            val j = map[i]
            j?.run {
                return@twoSum intArrayOf(j, index)
            }
            map.put(target-i, index)
        }
        return intArrayOf()
    }
}

fun main() {
    println(Solution().twoSum(intArrayOf(2,7,11,13), 9).map { it })
}